sdk/data/azappconfig/client.go

```diff
@@ -275,3 +275,288 @@ func (c *Client) NewListSettingsPager(selector SettingSelector, options *ListSet
 		Tracer: c.appConfigClient.Tracer(),
 	})
 }
+
+// NewGetSnapshotsPager - Gets a list of key-value snapshots.
+//
+//   - options - NewGetSnapshotsPagerOptions contains the optional parameters to retrieve a snapshot
+//     method.
+func (c *Client) NewGetSnapshotsPager(options *NewGetSnapshotsPagerOptions) *runtime.Pager[ListGetSnapshotsPagerResponse] {
+	opts := (*generated.AzureAppConfigurationClientGetSnapshotsOptions)(options)
+	ssRespPager := c.appConfigClient.NewGetSnapshotsPager(opts)
+
+	return runtime.NewPager(runtime.PagingHandler[ListGetSnapshotsPagerResponse]{
+		More: func(ListGetSnapshotsPagerResponse) bool {
+			return ssRespPager.More()
+		},
+		Fetcher: func(ctx context.Context, cur *ListGetSnapshotsPagerResponse) (ListGetSnapshotsPagerResponse, error) {
+			page, err := ssRespPager.NextPage(ctx)
+			if err != nil {
+				return ListGetSnapshotsPagerResponse{}, err
+			}
+
+			snapshots := []Snapshot{}
+
+			for _, snapshot := range page.Items {
+
+				convertedETag := azcore.ETag(*snapshot.Etag)
+
+				var convertedFilters []KeyValueFilter
+
+				for _, filter := range snapshot.Filters {
+					convertedFilters = append(convertedFilters, KeyValueFilter{
```

Comment:
Since we already have the length of filters to convert, we can create the slice with the proper size and populate it.
```go
convertedFilters := make([]KeyValueFilter, len(snapshot.Filters))
for i := 0; i < len(snapshot.Filters); i++ {
	convertedFilters[i] = KeyValueFilter{
		Key:   snapshot.Filters[i].Key,
		Label: snapshot.Filters[i].Label,
	}
}
```